### PR TITLE
update tool versions and fix errors and warnings that originated from the update

### PR DESCRIPTION
**Pull request by hj-johannes-lee**
Signed-off-by: Hyeongju Johannes Lee <hyeongju.lee@intel.com>

Update tool versions
Fix the errors and warnings originated from the update:
-Correct type deviceInfo (->DeviceInfo) to make it public
-Fix gpu_plugin.go and vpu_plugin_test.go where stylecheck errors occur
-Fix deprecation warnings
-Rename type 'PatcherManager' to 'Manager' to solve exported errors
-Rename type 'SgxMutator' to 'Mutator' to solve exported errors

### CODE REVIEW

**Comment by rojkov** [comment]
Thanks!

Could you also update `.github/workflows/ci.yaml` accordingly? This change will trigger new linter warnings. They need to be fixed too.

**Review by mythi — CHANGES_REQUESTED** [review]
@rojkov while the checks are green, there are deprecation warnings coming out of `make lint`. We'll fix those too before merging (marking as 'request changes' to block merging)

File: cmd/gpu_plugin/rm/gpu_plugin_resource_manager.go
Comment (mythi) [inline cmd/gpu_plugin/rm/gpu_plugin_resource_manager.go:68]: @uniemimu PTAL. the linter error was that `NewDeviceInfo` returns unexported type. (revive/unexported-return)

**Review by mythi — APPROVED** [review]
Looks good to me, thanks

**Review by rojkov — APPROVED** [review]
Thanks! This looks perfect!

**Pull request merged**
